### PR TITLE
Modularize lengths

### PR DESCRIPTION
**Pull request by meg-huggingface**
Removes the text lengths logic into its own dedicated library in data_measurements/lengths/ . In so doing, makes it much more streamlined. Mirrors the other "modularization" we have now done for labels, npmi, duplicates, etc.

### CODE REVIEW

**Review by sashavor — APPROVED** [review]
Some very minor comments/typos, otherwise looks great, @meg-huggingface !

File: data_measurements/text_duplicates/text_duplicates.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def _load_duplicates_cache(self):
         if exists(self.dups_result_json_fid):
             results = ds_utils.read_json(self.dups_result_json_fid)
         return results
-    
+
     def _write_duplicates_cache(self):
         """Writes newly computer results to cache."""
```
[the remark this answers is not in the document]
Comment (meg-huggingface) [reply data_measurements/text_duplicates/text_duplicates.py]: Thanks, done!

File: run_data_measurements.py [hunk 2]
```diff
@@ -88,9 +88,11 @@ def load_or_prepare(dataset_args, calculation=False, use_cache=False):
     if do_all or calculation == "lengths":
         logs.info("\n* Calculating text lengths.")
         dstats.load_or_prepare_text_lengths()
-        logs.info("Done!")
-        logs.info(
-            "- Text length results now available at %s." % dstats.length_df_fid)
+        length_fid_dict = dstats.length_obj.get_filenames()
+        print("If all went well, then results are in the following files:")
```
[the remark this answers is not in the document]
Comment (meg-huggingface) [reply run_data_measurements.py:92]: Hmm good idea. Can add in the next PR and have a similar logic for all of the modules.

File: data_measurements/dataset_statistics.py [hunk 3]
```diff
@@ -305,71 +324,14 @@ def load_or_prepare_text_lengths(self, load_only=False):
         a figure of the text lengths, some text length statistics, and
         a text length dataframe to peruse.
         Args:
-            save:
+            load_only: Whether we can compute anew, or just need to try to grab cache.
```
[the remark this answers is not in the document]
Comment (meg-huggingface) [reply data_measurements/dataset_statistics.py:327]: Done.

**Pull request merged**